average.js

```diff
@@ -1,6 +1,17 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  
+  let invalidData = 0;
+  
+  return parseFloat(numbers.reduce((p, c)=> {
```

Comment:
Good translation of requirement into a single-pass loop. This solution is good in many cases.
Be aware of the trade-off as well: counting the NaN in one loop and summing up the numbers in another allows them to execute in parallel, potentially reducing the lead-time when the amount of data is huge.

Reply:
Hello @numocityadmin, thank you very much for your reply. Thanks for letting me know this problem. I thought about this problem and came up with a solution in which we can first filter out the NaN values and then calculate the average.

Reply:
@numocityadmin  I have made the changes please look into it once and let me know your views about it.